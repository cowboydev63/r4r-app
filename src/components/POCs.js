import React from 'react';
import PropTypes from 'prop-types';
import ContactInformation from './ContactInformation';
import { poCsInterface } from '../interfaces';

const POCs = ({ poCs }) => {
    if(!poCs.length) {
        return null;
    }
    return (
        <article>
            <h2>Contact Information</h2>
            { 
                poCs.map((poc, idx) => (
                    <ContactInformation contact={ poc } key={ idx } />
                ))
            }
        </article>
    )
}

POCs.propTypes = {
    poCs: PropTypes.arrayOf(poCsInterface)
}

POCs.defaultProps = {
    poCs: [],
}

export default POCs;