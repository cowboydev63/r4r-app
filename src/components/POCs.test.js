import React from 'react';
import { shallow } from 'enzyme';
import POCs from './POCs';

describe('POCs Component', () => {
  it('renders correctly without poC', () => {
    const wrapper = shallow(<POCs />);
    expect(wrapper).toMatchSnapshot();
})

it('renders correctly with poC', () => {
    const example = {
        name: {
            prefix: 'Mr',
            firstName: 'Franklin',
            middleName: 'Delano',
            lastName: 'Roosevelt',
            suffix: 'III',
        },
        title: 'Senor',
        phone: '123-555-5555',
        email: 'waitingforgodot@gmail.com',
    }
    const wrapper = shallow(<POCs poCs={[example]}/>)
    expect(wrapper).toMatchSnapshot();
  })
})