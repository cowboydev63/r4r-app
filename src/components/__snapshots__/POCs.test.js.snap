// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`POCs Component renders correctly with poC 1`] = `
<article>
  <h2>
    Contact Information
  </h2>
  <ContactInformation
    contact={
      Object {
        "email": "waitingforgodot@gmail.com",
        "name": Object {
          "firstName": "Franklin",
          "lastName": "Roosevelt",
          "middleName": "Delano",
          "prefix": "Mr",
          "suffix": "III",
        },
        "phone": "123-555-5555",
        "title": "Senor",
      }
    }
    key="0"
  />
</article>
`;

exports[`POCs Component renders correctly without poC 1`] = `""`;
